ghost/adapter-cache-fs/src/AdapterCacheFS.ts

```diff
@@ -0,0 +1,107 @@
+import crypto from 'crypto';
+import fs from 'fs-extra';
+import path from 'path';
+import Base from '@tryghost/adapter-base-cache';
+
+type AdapterOptions = {
+    cachePath?: string;
+}
+
+/**
+ * Filesystem-based cache adapter for Ghost
+ *
+ * To instantly invalidate the cache, we use a timestamp in the folder name. This means that
+ * when we want to invalidate the cache, we can just change the timestamp and we drop all knowledge
+ * of the old cache.
+ *
+ * To keep the overhead low, we only create the directory structure upon initialization, and then
+ * during the reset() method. ie. we don't do this for ever set() call. If we want to do further directory
+ * partitioning, this may need to change.
+ *
+ * It is up to the host system to clean up files periodically.
+ */
+export default class AdapterCacheFS extends Base {
+    // Keep a reset count to protect against resets occuring in the same millisecond and the cache not
```

Comment:
Good thinking - though we wouldn't have this issue if we're to use a "random" string rather than a timestamp for the folder name? I realise we lose a simple way to see the most recent entries, but the filesystem will have that data 🤔